Index: libelfsh/reloc.c
===================================================================
--- libelfsh/reloc.c	(revision 1446)
+++ libelfsh/reloc.c	(working copy)
@@ -194,7 +194,7 @@
   if (r == NULL)
     PROFILER_ERR(__FILE__, __FUNCTION__, __LINE__, 
 		      "Invalid NULL parameter", -1);
-  r->r_addend = (elfsh_Sword) val;
+  r->r_addend = val;
   PROFILER_ROUT(__FILE__, __FUNCTION__, __LINE__, 0);
 }
 
